src/sentry/migrations/0726_backfill_groupsearchviews_with_pinned_searches.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 5.0.6 on 2024-06-03 17:27
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_groupsearchviews_with_pinned_searches(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    GroupSearchView = apps.get_model("sentry", "GroupSearchView")
+    SavedSearch = apps.get_model("sentry", "SavedSearch")
+
+    pinned_searches = SavedSearch.objects.filter(visibility="owner_pinned")
```

Comment:
This table is pretty small in saas, so the table scan here should be ok. If it isn't you can remove the condition from the query and implement it as a condition in python code.

Reply:
Gotcha. Just out of curiosity, what is the concern with having the condition within the query instead of just in code? Is it just faster for python to loop through all of the records with an if statement instead of having Postgres query  for it?

Reply:
As well as this, sometimes even if you have an index, when you use `RangeQuerySetWrapper` it adds `where id > x order by id` to all queries. So unless the index also includes `id`as the last column you can also run into problems. Iterating over the whole table takes longer as a whole, but each individual query is pretty cheap